buildSrc/src/main/java/org/elasticsearch/gradle/internal/CheckRestCompatPlugin.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.gradle.internal;
+
+import org.elasticsearch.gradle.ElasticsearchJavaPlugin;
+import org.gradle.api.Plugin;
+import org.gradle.api.Project;
+import org.gradle.api.Task;
+import org.gradle.api.plugins.JavaBasePlugin;
+import org.gradle.api.tasks.TaskProvider;
+
+/**
+ * Lifecycle plugin to anchor all Rest compatibility testing to a single command. Usage: `checkRestCompat`
+ */
+public class CheckRestCompatPlugin implements Plugin<Project> {
```

Comment:
@jakelandis any strong motivation for this being a separate plugin rather than just folding this into `YamlRestCompatTestPlugin`. I have no issue with plugins like that _also_ implementing lifecycle conventions.

Reply:
nope ... no strong motivation just thought lifecycle plugins would be more standalone entities. I can move the task over.

Reply:
done on https://github.com/elastic/elasticsearch/pull/67394/commits/81b333076f8d6a3ee98d9ba01740995b0a8ad8a6